### ISSUE TITLE
XRP asset is not correctly encoded on XAHAU

### ISSUE THREAD

**Issue by tadejgolobic** [WietseWind/xrpl-accountlib#313]
When you are on non-xrp native ledger / sidechain, XRP currency is encoded as 00000...00. However, 0000...00 represents native currency (on Xahau network, that is XAH) adn thus transaction fails.

Imagine following scenario
```
{
    "TransactionType": "TrustSet",
    "Account": "ra5nK24KXen9AHvsdFTKHSANinZseWnPcX",
    "LimitAmount": {
      "currency": "XRP",
      "issuer": "rsP3mgGb2tcYUrxiLFiHJiQXhsziegtwBc",
      "value": "100"
    },
}
```

on Xahau chain, this will actually try to make trustline to above issuer for currency XAH. In order to create / use XRP in IOU way, you need to manually specify a long 40 nibble currency code with 'X' 'R' 'P' (0000000000000000000000005852500000000000) and then it is working.

I am also attaching binary visualizer of one of my trustset transaction (it is not the above transaction). Note the currency code, even though XRP was specified as currency.
![image](https://github.com/WietseWind/xrpl-accountlib/assets/60660023/834e4ebd-b36c-4431-a132-55953daae977)

**Comment by WietseWind** [comment]
Good point! 

Adding now :)

**Comment by tadejgolobic** [comment]
Yes, the perfect way to implement this would be probably in `ripple-binary-codec` however, that can probably take some time. This is why I opened issue here.

Not really sure what would be the best way to implement this in this library. I used it for signing transaction offline, and then copying transaction blob to online computer and submitting transaction there. This was my use case if it helps :)

**Comment by WietseWind** [comment]
@tadejgolobic Implemented and added to our custom repo used by `xrpl-accountlib`

1. Fix in binary codec: https://github.com/XRPLF/xrpl.js/commit/94ad22162d39a8badf56a5d7beb634b1f96eef51
2. Fix in accountlib: https://github.com/WietseWind/xrpl-accountlib/commit/a1661668f963839d7c9211d9da3e8fd7f6b3ed1a

This is now published in `xrpl-accountlib@2.5.1` - https://www.npmjs.com/package/xrpl-accountlib/v/2.5.1

You can use it like this:

```javascript
// test.mjs

import { generate, sign, nativeAsset } from 'xrpl-accountlib'

const account = generate.familySeed({ algorithm: 'ed25519' })

const tx = {
  TransactionType: "TrustSet",
  Account: "ra5nK24KXen9AHvsdFTKHSANinZseWnPcX",
  LimitAmount: {
    currency: "XRP",
    issuer: "rsP3mgGb2tcYUrxiLFiHJiQXhsziegtwBc",
    value: "100"
  },
}

const c1 = sign(tx, account).signedTransaction.slice(24, 24+40)
console.log(c1)

nativeAsset.set('XAH')
const c2 = sign(tx, account).signedTransaction.slice(24, 24+40)
console.log(c2)

tx.LimitAmount.currency = 'XAH'
const c3 = sign(tx, account).signedTransaction.slice(24, 24+40)
console.log(c3)
```

Output:
```
0000000000000000000000000000000000000000
0000000000000000000000005852500000000000
0000000000000000000000000000000000000000
```
